boilerplate/frontend/content/main.js

```diff
@@ -1,3 +1,22 @@
+const { XPCOMUtils } = ChromeUtils.importESModule(
+  "resource://gre/modules/XPCOMUtils.sys.mjs"
+);
+
+XPCOMUtils.defineLazyModuleGetters(this, {
+  DevtoolsServer: "resource:///modules/DevtoolsServer.jsm",
+});
+
 function showMore() {
   document.getElementById("more-text").hidden = false;
 }
+
+function startDevtools() {
+  const devtools = new DevtoolsServer();
```

Comment:
Two things:

- What prevents the user from clicking on this twice?  Once you've started a devtools server, you shouldn't want to start another.
- Have you considered having a static property on the DevtoolsServer which is an instance you can refer to, or maybe a static method which returns a singleton on demand?  (This might solve the first problem here as well.)

Reply:
I have made some changes:
 - It is now a singleton which you should get an instance of with the `get()` method
 - The constructor is marked as private and will throw an error if called twice, but because it is javascript anyone can still call it
 - I have added a few checks so functions that should only be called once will warn and exit if called repeatedly (i.e. `start`)